app/controllers/posts_controller.rb

```diff
@@ -36,8 +38,24 @@ def markdown_num
     if params[:revision].present?
       post_revision = find_post_revision_from_topic_id
       render plain: post_revision.modifications[:raw].last
+    elsif params[:post_number].present?
+      markdown Post.find_by(topic_id: params[:topic_id].to_i, post_number: params[:post_number].to_i)
     else
-      markdown Post.find_by(topic_id: params[:topic_id].to_i, post_number: (params[:post_number] || 1).to_i)
+      topic = Topic.find_by(id: params[:topic_id].to_i)
+      raise Discourse::NotFound unless guardian.can_see?(topic)
+      offset = [((params[:page] || 1).to_i * MARKDOWN_TOPIC_PAGE_SIZE) - MARKDOWN_TOPIC_PAGE_SIZE, 0].max
+      posts = topic.posts.order(:post_number).limit(MARKDOWN_TOPIC_PAGE_SIZE).offset(offset)
```

Comment:
I think we should use a topic view to be safe here. Otherwise I'd worry about including posts that should not be seen such as moderator notes, etc.